js/dll.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+
+class Node{
+  constructor(val, next = null, prev = null){
+    this.val = val
+    this.next = next
+    this.prev = prev
+  }
+}
+
+class DLL{
+  constructor(){
+    this.head = null
+    this.tail = null
+    this.len = 0
+  }
+
+  push(val){
+    let oldHead = this.head
+    this.head = new Node(val, this.next = this.head)
+    this.len ++
+    if(oldHead){
+      oldHead.prev = this.head
+    }else{
+      this.tail = this.head
+    }
+  }
+
+  append(val){
+    let oldTail = this.tail
+    this.tail = new Node(val, this.prev = this.tail)
+    this.len ++
+    if(oldTail){
+      oldTail.next = this.tail
+    }else{
+      this.head = this.tail
+    }
+  }
+
+  pop(){
+    if(this.len === 0){
+      this.head = null
+      this.tail = null
+    }
+    if(!this.head){
+      throw new Error('Node not in list.')
```

Comment:
Is this really the error you want to throw here? And when would this get hit? The head is only null when the length is 0